src/Mailer/Transport/SmtpTransport.php

```diff
@@ -214,6 +221,36 @@ protected function _bufferResponseLines(array $responseLines): void
         $this->_lastResponse = array_merge($this->_lastResponse, $response);
     }
 
+    /**
+     * Parses the last response line and extract the preferred authentication type.
+     *
+     * @return void
+     */
+    protected function _parseAuthType(): void
+    {
+        $this->authType = null;
+
+        $auth = '';
+        foreach ($this->_lastResponse as $line) {
+            if (empty($line['message']) || substr($line['message'], 0, 5) === 'AUTH ') {
```

Comment:
```suggestion
            if (strlen($line['message']) == 0 || substr($line['message'], 0, 5) === 'AUTH ') {
```

empty() will also accept '0' which wouldn't be a good auth type. Using strlen() will be safer.